R/pkg/R/serialize.R

```diff
@@ -36,188 +36,233 @@
+writeObject.default = function(object, con, writeType = TRUE) {
+  stop(paste("Unsupported type for serialization", class(object)))
+}
 
-  serdeType <- getSerdeType(object)
+# integer same as logical; will cast TRUE -> 1, FALSE -> 0
+writeObject.integer <-
+writeObject.logical <- function(object, con, writeType = TRUE) {
   if (writeType) {
-    writeType(con, serdeType)
+    writeType(object, con)
   }
-  switch(serdeType,
-         NULL = writeVoid(con),
-         integer = writeInt(con, object),
-         character = writeString(con, object),
-         logical = writeBoolean(con, object),
-         double = writeDouble(con, object),
-         numeric = writeDouble(con, object),
-         raw = writeRaw(con, object),
-         array = writeArray(con, object),
-         list = writeList(con, object),
-         struct = writeList(con, object),
-         jobj = writeJobj(con, object),
-         environment = writeEnv(con, object),
-         Date = writeDate(con, object),
-         POSIXlt = writeTime(con, object),
-         POSIXct = writeTime(con, object),
-         stop(paste("Unsupported type for serialization", type)))
-}
+  # non-scalar value written as array
+  if (length(object) > 1L) return(writeArray(object, con))
+  if (is.na(object)) return() # no value for NULL
 
-writeVoid <- function(con) {
-  # no value for NULL
+  writeBin(as.integer(object), con, endian = "big")
 }
-
-writeJobj <- function(con, value) {
-  if (!isValidJobj(value)) {
-    stop("invalid jobj ", value$id)
+writeObject.character <- function(object, con, writeType = TRUE) {
+  if (writeType) {
+    writeType(object, con)
   }
-  writeString(con, value$id)
-}
+  # non-scalar value written as array
+  if (length(object) > 1L) return(writeArray(object, con))
+  if (is.na(object)) return() # no value for NULL
 
-writeString <- function(con, value) {
   utfVal <- enc2utf8(value)
-  writeInt(con, as.integer(nchar(utfVal, type = "bytes") + 1))
+  writeObject(as.integer(nchar(utfVal, type = "bytes") + 1L), writeType = FALSE)
   writeBin(utfVal, con, endian = "big", useBytes = TRUE)
 }
+writeObject.numeric <- function(object, con, writeType = TRUE) {
+  if (writeType) {
+    writeType(object, con)
+  }
+  # non-scalar value written as array
+  if (length(object) > 1L) return(writeArray(object, con))
+  if (is.na(object)) return() # no value for NULL
 
-writeInt <- function(con, value) {
-  writeBin(as.integer(value), con, endian = "big")
+  writeBin(object, con, endian = "big")
 }
-
-writeDouble <- function(con, value) {
-  writeBin(value, con, endian = "big")
+writeObject.raw <- function(object, con, writeType = TRUE) {
+  if (writeType) {
+    writeType(object, con)
+  }
+  writeObject(length(object), con, writeType = FALSE)
+  writeBin(object, con, endian = "big")
 }
-
-writeBoolean <- function(con, value) {
-  # TRUE becomes 1, FALSE becomes 0
-  writeInt(con, as.integer(value))
+writeType.list = function(object, con) {
```

Comment:
ditto

Reply:
all other places

Reply:
Yep, have noted the style feedback from Jenkins. Will do when i get a chance.

About the "Remove comment" feedbacks, it's just line-final comments that are out-of-style? each comment should get its own line?